src/main/java/com/tikal/jenkins/plugins/multijob/views/MultiJobView.java

```diff
@@ -170,25 +172,24 @@ private void addProjectFromBuilder(MultiJobProject project,
                 if( project.getLastBuild() != null ) {
                     for ( MultiJobBuild mjb : project.getBuilds() ) {
                     	for ( SubBuild sb : mjb.getSubBuilds() ) {
-							if (sb.getJobAlias() != null) {
-								if (sb.getJobAlias().equals(projectConfig.getJobAlias())) {
-									tmp_build = subProject.getBuildByNumber(sb.getBuildNumber());
-									if (tmp_build != null) {
-										if (latestAliasBuild == null) {
-											latestAliasBuild = tmp_build;
-										}
-
-										if (lastFailure == 0 && tmp_build.getIconColor() == BallColor.RED) {
-											lastFailure = tmp_build.getNumber();
-										} else if (lastSuccess == 0 && tmp_build.getIconColor() == BallColor.BLUE) {
-											lastSuccess = tmp_build.getNumber();
-										}
-
-										if (lastFailure != 0 && lastSuccess != 0) {
-											break;
-										}
-									}
-								}
+                            if (sb.getJobAlias().equals(projectConfig.getJobAlias()) &&
+                                    sb.getJobName().equals(projectConfig.getJobName())) {
+                                tmp_build = subProject.getBuildByNumber(sb.getBuildNumber());
+                                if (tmp_build != null) {
```

Comment:
Technically speaking, I do not like this deeply nested if.
It would be much easier to follow in the following way:

```
if (tmp_build == null) {
  continue;
}
// other checks
```